demosplan/DemosPlanCoreBundle/Command/Data/GenerateCustomerCommand.php

```diff
@@ -13,53 +13,94 @@
 namespace demosplan\DemosPlanCoreBundle\Command\Data;
 
 use demosplan\DemosPlanCoreBundle\Command\CoreCommand;
-use demosplan\DemosPlanCoreBundle\Entity\User\Customer;
 use demosplan\DemosPlanCoreBundle\Exception\EntryAlreadyExistsException;
-use demosplan\DemosPlanUserBundle\Repository\CustomerRepository;
+use demosplan\DemosPlanUserBundle\Logic\CustomerService;
+use Doctrine\ORM\EntityManagerInterface;
 use Exception;
+use InvalidArgumentException;
 use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Helper\QuestionHelper;
 use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Question\Question;
 use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
 
+use function in_array;
+use function is_string;
+
 class GenerateCustomerCommand extends CoreCommand
 {
+    private const OPTION_NAME = 'name';
+    private const OPTION_SUBDOMAIN = 'subdomain';
+
     protected static $defaultName = 'dplan:data:generate-customer';
     protected static $defaultDescription = 'Creates a new customer';
+
+    protected QuestionHelper $helper;
+
     /**
-     * @var QuestionHelper
+     * @var list<string>
      */
-    protected $helper;
+    private array $reservedNames;
+
     /**
-     * @var CustomerRepository
+     * @var list<string>
      */
-    private $customerRepository;
+    private array $reservedSubdomains;
 
     public function __construct(
-        CustomerRepository $customerRepository,
+        private readonly CustomerService $customerService,
+        private readonly EntityManagerInterface $entityManager,
         ParameterBagInterface $parameterBag,
         string $name = null
     ) {
         parent::__construct($parameterBag, $name);
-        $this->customerRepository = $customerRepository;
         $this->helper = new QuestionHelper();
+        $reservedCustomers = $this->customerService->getReservedCustomerNamesAndSubdomains();
+        $this->reservedNames = array_column($reservedCustomers, 0);
+        $this->reservedSubdomains = array_column($reservedCustomers, 1);
+    }
+
+    protected function configure(): void
+    {
+        parent::configure();
+        $this->addOption(
+            self::OPTION_NAME,
+            'i',
```

Comment:
why `i`?

Reply:
`n` was taken for some reason, `i` didn't feel worse than any of the other 24 choices.